test/SILOptimizer/existential_specializer_soletype.sil

```diff
@@ -0,0 +1,138 @@
+// RUN: %target-sil-opt  -wmo -enable-sil-verify-all %s -enable-sil-existential-specializer -existential-specializer  -inline -sil-combine -generic-specializer -devirtualizer
+
+// This file tests existential specializer transformation followed by concrete type propagation  and generic specialization leading to a devirtualization of a witness method call.
+
+sil_stage canonical
+
+import Builtin
+import Swift
+import SwiftShims
+
+protocol RP {
+  func getThres() -> Int32
+}
+
+class RC : RP {
+  @inline(never) func getThres() -> Int32
+}
+
+sil hidden [noinline] @$s6simple2RCC8getThress5Int32VyF : $@convention(method) (@guaranteed RC) -> Int32 {
+bb0(%0 : $RC):
+  %1 = integer_literal $Builtin.Int32, 10
+  %2 = struct $Int32 (%1 : $Builtin.Int32)
+  return %2 : $Int32
+} 
+
+// CHECK-LABEL: sil hidden [noinline] @$s6simple12find_wrapperSbyF : $@convention(thin) () -> Bool {
+// CHECK: bb0:
+// CHECK:   alloc_ref
+// CHECK:   integer_literal
+// CHECK:   struct
+// CHECK:   alloc_stack
+// CHECK:   store
+// CHECK:   strong_retain
+// CHECK:   function_ref @$s6simple4find5count4Obj1Sbs5Int32V_AA2RP_ptFTf4ne_n4main2RCC_Tg5 : $@convention(thin) (Int32, @guaranteed RC) -> Bool
```

Comment:
this function signature is checked that has been generic specialized after existential specialization. The original function took a protocol as argument, now it is the concrete type.

Reply:
Done